### PR TITLE
add catalog subcmd

### PR DESCRIPTION
**Pull request by bharathkkb**
This adds a catalog subcommand for exploring blueprints. Currently it only lists available blueprints and some basic info. In the future once metadata rolls out, this will be a client consuming that info.

b/254514965

### CODE REVIEW

File: cli/bpcatalog/cmd.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/spf13/viper"
+)
+
+var catalogListFlags struct {
+	format renderFormat
+	sort   sortOption
+}
+
+const (
+	tgmOrg = "terraform-google-modules"
```
Comment (g-awmalik) [inline cli/bpcatalog/cmd.go:17]: nit: can we call this tfModulesOrg

File: cli/bpcatalog/gh.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+func newGHService(opts ...ghServiceOption) *ghService {
+	ctx := context.Background()
+	ghs := &ghService{
+		client: github.NewClient(nil),
+		ctx:    ctx,
+	}
+	for _, opt := range opts {
+		opt(ghs)
+	}
```
Comment (g-awmalik) [inline cli/bpcatalog/gh.go:59]: Is this doing anything since the body for `type ghServiceOption` is empty?

File: cli/bpcatalog/tf_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+				},
+			},
+			want: []string{
+				"terraform-google-foo",
+				"terraform-google-bar",
+			},
+			sortBy: sortCreated,
+		},
+		{
+			name: "simple created",
```
Comment (g-awmalik) [inline cli/bpcatalog/tf_test.go:44]: all tests have the same name

**Review by g-awmalik — APPROVED** [review]
Minor nits.

File: cli/bpcatalog/tf_test.go (hunk 3 above)
Comment (bharathkkb) [reply cli/bpcatalog/tf_test.go]: fixed

File: cli/bpcatalog/cmd.go (hunk 1 above)
Comment (bharathkkb) [reply cli/bpcatalog/cmd.go]: done

File: cli/bpcatalog/gh.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
+package bpcatalog
```
[the remark this answers is not in the document]
Comment (bharathkkb) [reply cli/bpcatalog/gh.go:1]: We could possibly move. Does any thing else need this now? I err on the side of moving to utils unless we need this for some functionality elsewhere.

File: cli/bpcatalog/gh.go (hunk 2 above)
Comment (bharathkkb) [reply cli/bpcatalog/gh.go:59]: Not sure what you meant by empty - it is a type alias for func(*ghService) which can modify a struct of ghService.

**Pull request merged**